### PR TITLE
fix(eval): add the vimscript-1 feature to has()

### PR DESCRIPTION
**Pull request by seandewar**
Include `Test_string_concat_scriptversion1` (from v8.1.1116 and v8.1.1190) to test that
`has('vimscript-1')` works. (Hopefully that's OK)

Ref: https://github.com/neovim/neovim/pull/14611#issuecomment-921099818

### CODE REVIEW

**Review by justinmk — APPROVED** [review]

File: src/nvim/testdir/test_eval_stuff.vim [hunk 1]
```diff
@@ -117,6 +117,21 @@ func Test_skip_after_throw()
   endtry
 endfunc
 
+func Test_string_concat_scriptversion1()
```
Comment (janlazo) [inline src/nvim/testdir/test_eval_stuff.vim:120]: Add a comment `scriptversion 1` to match Vim.

**Review by janlazo — APPROVED** [review]

File: src/nvim/testdir/test_eval_stuff.vim [hunk 2]
```diff
@@ -117,6 +117,22 @@ func Test_skip_after_throw()
   endtry
 endfunc
 
+" scriptversion 1
```
Comment (janlazo) [inline src/nvim/testdir/test_eval_stuff.vim:120]: `:scriptversion` tests come after `Test_let_errmsg()`.

```vim
scriptversion 1
func Test_vvar_scriptversion1()
  call assert_equal(15, 017)
  call assert_equal(15, 0o17)
  call assert_equal(15, 0O17)
  call assert_equal(18, 018)
  call assert_equal(511, 0o777)
endfunc
``` 
is missing.

Comment (seandewar) [reply src/nvim/testdir/test_eval_stuff.vim:120]: I'll add it. :+1:

Comment (seandewar) [reply src/nvim/testdir/test_eval_stuff.vim]: A few of the functions in `test_eval_stuff.vim` seems to be in a different order compared to Vim.
I could re-arrange them all if you'd like to match Vim?

Comment (janlazo) [reply src/nvim/testdir/test_eval_stuff.vim]: Sure in a separate commit.

**Review by janlazo — APPROVED** [review]

**Pull request merged**